### PR TITLE
NV-2473 - data field is not added in case of inline of trigger

### PR DESCRIPTION
**Pull request by gitstart**
### What change does this PR introduce?
This fix ensures data field is added in case of inline of trigger

#### Description
We can create a new subscriber on the fly using inline of trigger

[https://docs.novu.co/platform/subscribers#2-inline-of-trigger](https://docs.novu.co/platform/subscribers#2-inline-of-trigger)

All attributes except data is updated/added in this case

 ### Why was this change needed?
closes https://github.com/novuhq/novu/issues/3905

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

**Comment by gitstart** [comment]
@jainpawan21 this PR is ready for review

File: packages/node/src/lib/events/events.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export class Events extends WithHttp implements IEvents {
       transactionId: data.transactionId,
       overrides: data.overrides || {},
       ...(data.actor && { actor: data.actor }),
+      data,
```
Comment (ainouzgali) [inline packages/node/src/lib/events/events.ts:20]: @gitstart this is not the right place for it. The data of the subscriber will be already included inside the data.to. 
The change you need to make will be inside the process-subscriber usecase. I believe it is not passed to the create subscriber there.

**Review by ainouzgali — COMMENTED** [review]
@gitstart could you also please add a test after you make the changes? :)

File: packages/node/src/lib/subscribers/subscribers.ts [hunk 2]
```diff
@@ -48,6 +48,12 @@ export class Subscribers extends WithHttp implements ISubscribers {
     });
   }
 
+  async create(subscriberId: string, data: ISubscriberPayload) {
```
Comment (ainouzgali) [inline packages/node/src/lib/subscribers/subscribers.ts:51]: @gitstart this is not needed. There is the identify function that does that. 
No changes are needed in the node sdk.

**Review by ainouzgali — COMMENTED** [review]
@gitstart thanks. Adding a test for this should be done in this [test](https://github.com/novuhq/novu/blob/6de3cf8f0ab1434a203866f4fda49a69336c7331/apps/api/src/app/events/e2e/trigger-event.e2e.ts#L258)

Please remove all changes in node sdk, they are not needed :)

File: packages/node/src/lib/subscribers/subscribers.ts (hunk 2 above)
Comment (gitstart) [reply packages/node/src/lib/subscribers/subscribers.ts]: The function has been removed @ainouzgali, thanks for the feedback.

**Review by ainouzgali — APPROVED** [review]
Thanks @gitstart !

**Pull request merged**